pkg/operator/controller/canary/http.go

```diff
@@ -0,0 +1,118 @@
+package canary
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"os"
+	"strings"
+	"time"
+
+	routev1 "github.com/openshift/api/route/v1"
+
+	"github.com/tcnksm/go-httpstat"
+)
+
+const (
+	echoServerPortAckHeader = "x-request-port"
+)
+
+// probeRouteEndpoint probes the given route's host
+// and returns an error when applicable.
+func probeRouteEndpoint(route *routev1.Route) error {
+	host := route.Spec.Host
+
+	if len(host) == 0 {
+		return fmt.Errorf("route.Spec.Host is empty, cannot test route")
+	}
+
+	// Create HTTP request
+	request, err := http.NewRequest("GET", "http://"+host, nil)
+	if err != nil {
+		return fmt.Errorf("error creating canary HTTP request: %v", err)
```

Comment:
log the request URL - might help any future diagnosis.

Reply:
fixed.